File: programming/lab2/src/main/java/moves/status/StringShot.java
package moves.status;

import ru.ifmo.se.pokemon.*;

/**
 * String Shot lowers the target's Speed by two stages.
 */
public class StringShot extends StatusMove {
    public StringShot() {
        this(95);
    }

    /**
     * Констурктор для тестов.
     * Необходим для 100% точности атаки
     * @param acc
     */
    public StringShot(double acc) {
        super(Type.BUG, 0, acc);
    }

    @Override
    protected boolean checkAccuracy(Pokemon pokemon, Pokemon pokemon1) {
        return (accuracy == 100) || super.checkAccuracy(pokemon, pokemon1);
    }

    @Override
    protected String describe() {
        return "использует String Shot";
    }

    @Override
    protected void applyOppEffects(Pokemon pokemon) {
        pokemon.addEffect(new Effect().stat(Stat.SPEED, -2).turns(-1));
    }
}

File: programming/lab2/app/src/main/kotlin/org/itmo/moves/status/StringShot.kt
package org.itmo.moves.status

import ru.ifmo.se.pokemon.*

class StringShot : StatusMove(Type.BUG, 0.0, 95.0) {
    override fun applyOppEffects(p0: Pokemon) {
        p0.addEffect(Effect().turns(-1).stat(Stat.SPEED, -2))
    }

    override fun describe(): String {
        return "использует String Shot"
    }
}